baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/DangerousRangeScanUsage.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = SeverityLevel.WARNING,
+        summary = "Should not use full atlasdb range scans")
+public final class DangerousRangeScanUsage extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+    private static final long serialVersionUID = 1L;
+
+    private static final Matcher<ExpressionTree> ALL_RANGE_METHOD_MATCHER = MethodMatchers.staticMethod()
+            .onClass("com.palantir.atlasdb.keyvalue.api.RangeRequest")
+            .named("all")
+            .withNoParameters();
```

Comment:
Since we own the source for this type, perhaps we should deprecate the method and add something like `@DoNotCall("Do not use full atlasdb range scans")` instead of adding a new custom error-prone rule. What do you think?

Reply:
there's also `@RestrictedApi` if you need to allowlist usages within Atlas

Reply:
@felixdesouza - what is this magic you speak of (RestrictedApi)? So wish I'd known about that last winter with the massive deprecations we did.

Reply:
haha yeah. It's this: https://errorprone.info/bugpattern/RestrictedApi

Reply:
[javadoc](https://errorprone.info/api/latest/com/google/errorprone/annotations/RestrictedApi.html) is better explainer